disnake/i18n.py

```diff
@@ -0,0 +1,232 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2021-present Disnake Development
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+import logging
+import os
+import warnings
+from collections import defaultdict
+from pathlib import Path
+from typing import DefaultDict, Dict, Optional, Set, Union
+
+from . import utils
+from .custom_warnings import LocalizationWarning
+from .enums import Locale
+
+MISSING = utils.MISSING
+
+
+Localizations = Union[str, Dict[Locale, str], Dict[str, str]]
+
+
+__all__ = (
+    "is_valid_locale",
+    "LocalizationStore",
+)
+
+
+_log = logging.getLogger(__name__)
+
+
+def is_valid_locale(locale: str) -> bool:
```

Comment:
technically should be defined in utils, if this is being exported.

Reply:
3dd10ed479cee6642c2f3a63ee5fece3cb0c858b